### ISSUE TITLE
breaking change in 0.30 - uninformative error message in _map_blocks

### ISSUE THREAD

**Issue by aaronspring** [Ouranosinc/xclim#854]
* xclim version: 0.30
* Python version:
* Operating System:

### Description

example below used to work in 0.29.

I couldn't figure out the issue from Traceback: `ValueError: One or more of the specified variables cannot be found in this dataset` 
I didn't find a link from HISTORY to changing PR and also could find the changing PR from the [list of commits](https://github.com/Ouranosinc/xclim/commits/master). probably [this](https://github.com/Ouranosinc/xclim/commit/b956f9ca9cd4bcdbfd221bc75fa9a8562475b3bd) was the change

### What I Did

Maybe this example is stupid but basically that's my usage in climpred. Did you change the sdba API?
```python
import xclim
print(xclim.__version__)
import xarray as xr

ds = xr.tutorial.load_dataset("rasm").isel(x=100,y=100)['Tair']

from xclim.sdba import EmpiricalQuantileMapping

reference = ds.isel(time=slice(None,12))
model = ds.isel(time=slice(12,24))
single_adjusted = EmpiricalQuantileMapping.train(reference, model, group='time').adjust(model)```
```

### What I Received

```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-8-c35cc135c214> in <module>
     10 reference = ds.isel(time=slice(None,12))
     11 model = ds.isel(time=slice(12,24))
---> 12 single_adjusted = EmpiricalQuantileMapping.train(reference, model, group='time').adjust(model)

~/anaconda3/envs/climpred-dev/lib/python3.9/site-packages/xclim/sdba/adjustment.py in train(cls, ref, hist, **kwargs)
    156         hist = convert_units_to(hist, ref)
    157 
--> 158         ds, params = cls._train(ref, hist, **kwargs)
    159         obj = cls(
    160             _trained=True,

~/anaconda3/envs/climpred-dev/lib/python3.9/site-packages/xclim/sdba/adjustment.py in _train(cls, ref, hist, nquantiles, kind, group)
    326             quantiles = nquantiles
    327 
--> 328         ds = eqm_train(
    329             xr.Dataset({"ref": ref, "hist": hist}),
    330             group=group,

<boltons.funcutils.FunctionBuilder-594> in _map_blocks(ds, **kwargs)

~/anaconda3/envs/climpred-dev/lib/python3.9/site-packages/xclim/sdba/base.py in _parse_group(*args, **kwargs)
    437     def _parse_group(*args, **kwargs):
    438         kwargs = _update_kwargs(kwargs)
--> 439         return func(*args, **kwargs)
    440 
    441     return _parse_group

~/anaconda3/envs/climpred-dev/lib/python3.9/site-packages/xclim/sdba/base.py in _map_blocks(ds, **kwargs)
    640             }
    641             ds = ds.drop_vars(extra_coords.keys())
--> 642             tmpl = tmpl.drop_vars(extra_coords.keys())
    643 
    644             # Call

~/anaconda3/envs/climpred-dev/lib/python3.9/site-packages/xarray/core/dataset.py in drop_vars(self, names, errors)
   4332             names = set(names)
   4333         if errors == "raise":
-> 4334             self._assert_all_in_dataset(names)
   4335 
   4336         variables = {k: v for k, v in self._variables.items() if k not in names}

~/anaconda3/envs/climpred-dev/lib/python3.9/site-packages/xarray/core/dataset.py in _assert_all_in_dataset(self, names, virtual_okay)
   4302             bad_names -= self.virtual_variables
   4303         if bad_names:
-> 4304             raise ValueError(
   4305                 "One or more of the specified variables "
   4306                 "cannot be found in this dataset"

ValueError: One or more of the specified variables cannot be found in this dataset
```

**Comment by aulemahal** [comment]
The auxiliary coordinate saga continues! The bug originates from the leftover `xc` and `yc`. Quick fix is to add `drop=True` to the `isel` call.

Thanks for the issue. I'm sorry to have introduced this bug... In some cases (not your I presume), auxiliary coordinates were giving me trouble. They are useless in most (all?) cases so I decided to simply drop them, and add them back after the wrapped function call. It also saves some useless dask tasks in some cases. But I didn't think of the case here : _scalar_ auxiliary coordinates.  I'll try to push a fix quickly, we could release a 0.30.1 soon.

Also, I note from your comment that we maybe should link PRs and issues in the entries of HISTORY.

**Comment by aaronspring** [comment]
> Also, I note from your comment that we maybe should link PRs and issues in the entries of HISTORY.

squash merges, less rebasing in master (I mean less single commits outside of PRs) and linking PR numbers and issues in history.rst helps me as a user to find breaking changes in the code.

**Comment by aaronspring** [comment]
I fixed xclim to 0.29 in climpred for now.
